### PR TITLE
br.storage: add delete api 

### PR DESCRIPTION
**Pull request by ben1009**
### What problem does this PR solve?
add delete api in br storage to support delete file in s3, local file, gcs 

Problem Summary:

### What is changed and how it works?

add delete api in br storage to support delete file 

What's Changed:

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/gcs.go [hunk 1]
```diff
@@ -88,6 +88,11 @@ type gcsStorage struct {
 	bucket *storage.BucketHandle
 }
 
+// DeleteFile delete the file in storage
+func (s *gcsStorage) DeleteFile(ctx context.Context, name string) error {
+	return errors.New("not implemented")
```
Comment (kennytm) [inline br/pkg/storage/gcs.go:93]: ... just implement this? you just call [`(*ObjectHandle).Delete`](https://pkg.go.dev/cloud.google.com/go/storage#ObjectHandle.Delete).

**Review by kennytm — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**